### ISSUE TITLE
Automatic logging of spans other than DB queries are not being done

### ISSUE THREAD

**Issue by amcsi** [nipwaayoni/elastic-apm-laravel#18]
This library used to record things like app boot and route match. It doesn't seem to anymore.

Here is an old screenshot of what used to be logged:

![2020-06-22_12-33](https://user-images.githubusercontent.com/116494/85296426-a9d0f700-b4a1-11ea-8309-adfafece37d3.png)

As a side note, as per the screenshot it appears like there used to be a problem with spans spanning past the primary transaction span even in the old code (#15)

**Comment by dstepe** [comment]
The screenshot didn't upload correctly. I'm happy to look into this. I worked on the previous stable release of the package, but there were changes in the pre-release which I based this on and I'm still getting familiar with it.

**Comment by amcsi** [comment]
@dstepe oops. Fixed.

**Comment by dstepe** [comment]
I looked back at the v5.8.2 release which I believe was considered stable and don't see anywhere that those Spans are created. Is it possible you were using customized middleware? If so, share it and I can look to incorporate those features. If not, can you determine what version of the agent was used to generate that data?

**Comment by amcsi** [comment]
@dstepe ah, you got me. It may have been this other fork of elastic-apm-laravel that I was using:
https://github.com/arkaitzgarro/elastic-apm-laravel/blame/49e777a02cc9cefa041252610746793d5e29ddfd/src/Collectors/FrameworkCollector.php#L25

I was a bit desperate finding a combination of elastic-apm-laravel and elastic-apm-php-agent that worked together to be honest, and I got confused such that I didn't really keep track of which fork's which version did what :P

**Comment by dstepe** [comment]
That other package looks very interesting. I'm going to reach out to the author about collaborating. Thanks for the pointer. I'll leave this issue open pending that conversation.

**Comment by dstepe** [comment]
@amcsi I'm curious about your experience with the other Laravel package you referenced and your comment that you were trying to find a "combination of elastic-apm-laravel and elastic-apm-php-agent that worked together". Before I approach the author of the other project, I'd like to better understand your experience and what brought you back to this package. Can you provide a little more detail on the issues you ran into?

**Comment by amcsi** [comment]
@dstepe When I first installed philkra/elastic-apm-laravel, I ran into philkra/elastic-apm-laravel#66; thus the package didn't work, it only crashed my project.
Then I tried to do what this person did in this PR: elastic-apm-laravel#67, but it didn't seem to help; I got the same crash due to `setSpans` being missing.

Then I found the arkaitzgarro fork of the package (maybe it was linked in one of the issues/PRs), and was able to that one running successfully.

There was nothing in particular from that fork that I wanted other than the simple thing that the library work successfully rather than crash.
